### ISSUE TITLE
emit non-provided enums whose type is used in a provided class

### ISSUE THREAD

**Issue by rkirov** [angular/clutz#188]
example

```
goog.provide('Foo');

/** @enum {string} */
somenamespace.Enum = {
};

/** @constructor */
Foo = function() {};
/** @type {somenamespace.Enum} */
Foo.prototype.bar;
```

**Comment by mprobst** [comment]
@rkirov we could either find and emit these, or emit as `void` at the usage site, so the API remains unusable. WDYT?

**Comment by rkirov** [comment]
I will try to emit those, as closure's permissiveness in this case, means they are basically an explicit export. In general in closure, when one is using a symbol purely in type position, it is not goog.required, so it seems consistent not to 'provide' it.

Unfortunately, TS doesn't work this way, so we will have to fix it.

**Comment by rkirov** [comment]
fixed in d19a3a01b8dfe1607bc2772d5305059eea937f46
